main/src/ca/uwaterloo/flix/language/phase/typer/TypeConstraint.scala

```diff
@@ -50,6 +54,8 @@ sealed trait TypeConstraint {
     */
   def numVars: Int = this match {
     case TypeConstraint.Equality(tpe1, tpe2, _) => tpe1.typeVars.size + tpe2.typeVars.size
+    case TypeConstraint.EqJvmConstructor(mvar, clazz, tpes, _) => mvar.baseType.size // TODO INTEROP add all typeVars size of tpes
+    case TypeConstraint.EqJvmMethod(mvar, clazz, tpes, _, _) => mvar.baseType.size // TODO INTEROP add all typeVars size of tpes
     case TypeConstraint.Trait(_, tpe, _) => tpe.typeVars.size
```

Comment:
I will let @mlutze comment on the two things above.

Reply:
Here we are counting the variables, so not `baseType` but just `1`, because `mvar` is a variable.

Then for `tpes` we can sum the typeVars in all the types.

Reply:
It should be resolved.